android_loomo_ros/src/org/ros/android/android_loomo_ros/MainActivity.java

```diff
@@ -74,13 +77,20 @@
 
     private Queue<Long> mDepthStamps;
 
+    // loading params from yaml file
+    private final NodeParams params = Utils.loadParams();
+
     // Assumes that ROS master is a different machine, with a hard-coded ROS_MASTER_URI.
     // If you'd like to be able to select the URI in the app on startup, replace
     // super( , , ) with super( , ) to start a different version of RosActivity
-    public MainActivity() { super("LoomoROS", "LoomoROS", URI.create("http://192.168.42.134:11311/"));}
+//    public MainActivity() { super("LoomoROS", "LoomoROS", URI.create("http://192.168.42.134:11311/"));}
+
+    // Pull ROS_MASTER_URI from yaml file
+    public MainActivity() { super("LoomoROS", "LoomoROS", URI.create(Utils.loadParams().getMasterURI())); }
```

Comment:
should this just be params.getMasterURI() ? no need to loadParams() again?

Reply:
It's a cheeky workaround because you can't reference a class field until the supertype constructor is called.